models/notification.go

```diff
@@ -718,6 +719,42 @@ func getNotificationCount(e Engine, user *User, status NotificationStatus) (coun
 	return
 }
 
+// UserIDCount is a simple coalition of UserID and Count
+type UserIDCount struct {
+	UserID int64
+	Count  int64
+}
+
+// GetUIDsAndNotificationCounts between the two provided times
+func GetUIDsAndNotificationCounts(since, until timeutil.TimeStamp) ([]UserIDCount, error) {
+	sql := `SELECT user_id, count(*) AS count FROM notification ` +
+		`WHERE user_id IN (SELECT user_id FROM notification WHERE updated_unix >= ? AND ` +
+		`updated_unix < ?) AND status = ? GROUP BY user_id`
+
+	res, err := x.Query(sql, since, until, NotificationStatusUnread)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(res) == 0 {
+		return []UserIDCount{}, nil
+	}
+
+	uidCounts := make([]UserIDCount, len(res))
+	for i, result := range res {
+		uid, err := strconv.ParseInt(string(result["user_id"]), 10, 64)
```

Comment:
Why the conversion? Usually xorm does the job. 🤔

Reply:
```
func (engine *Engine) Query(sqlOrArgs ...interface{}) (resultsSlice []map[string][]byte, err error) {
```

Xorm does not do the job.

Reply:
@zeripath 
```go
var res []UserIDCount
x.SQL(sql).Find(&res)
```
```

Reply:
Thanks! I just couldn't find that!